### DISCUSSION TITLE
datalake-graphql-wrapper: A GraphQL API for presto/trino

### DISCUSSION THREAD

**Discussion by noxify** [hayes/pothos#840] [Show and tell]
Hi,

I'm happy to present you my latest project.

https://github.com/dbsystel/datalake-graphql-wrapper

It's a template which allows you to provide a GraphQL API for the existing tables in trino/presto.

The template includes an interactive cli to generate the needed code for pothos including sorting, paging, filtering.

We're using this template ( a bit extended) also at work and it's awesome how fast we can add new endpoints in a standarized way.

We plan to extend the template in the future with some HowTos to have it as generic as possible.

Any thoughts? What could be improved? 

Thanks!

**Comment by hayes** [comment]
Sorry I didn't see this sooner, this looks really cool!

Just added it to the list of resources on the docs site, thanks!

I don't have a ton of experience working with things like presto, but this seems like a very cool use of pothos.

**Reply by noxify** [reply]
Hi,

using pothos ( together with graphql-yoga ) was the best decision I ever made ;) 
There is a lot of pain behind the scenes but I was able to manage this.

We added also a mutation support which uses the generated interfaces from the query.
With some other tools we were able to generate the parquet files, upload them to our s3 storage and generate the external tables in the hive meta store.

Currently we're working on a file upload support.

Unfortunately we haven't found the time to update the github package with the latest changes - but it's on my todo list.
